src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -51,15 +51,14 @@ public Product findProductByIdOrThrowException(UUID productId) {
     }
 
     @Override
-    public void deleteProduct(UUID productId){
+    public ProductResponse deleteProduct(UUID productId){
         ifProductIsNotDeletedThrowException(productId);
         Product product = findProductByIdOrThrowException(productId);
         product.setDeleted(true);
-        productRepository.save(product);
+        return ProductMapper.toProductResponse(productRepository.save(product));
     }
 
-    @Override
-    public void ifProductIsNotDeletedThrowException(UUID productId){
+    private void ifProductIsNotDeletedThrowException(UUID productId){
         Product product = findProductByIdOrThrowException(productId);
         if (product.isDeleted()){
             throw new IllegalStateException("Product with id " + productId + " is already deleted.");
```

Comment:
Esto me hace ruido porque dentro del cuerpo del deleteProduct ya estamos usando el findProductByIdOrThrowException. Creo que podría optimizarse si acá pasamos directamente el producto en vez de pegarle a la base de datos de nuevo. Esto podría quedar así

```suggestion
    private void ifProductIsNotDeletedThrowException(Product product) {
        if (product.isDeleted()){
            throw new ResponseStatusException("Product with id " + productId + " is already deleted.");
```